sdk/python/kfp/components/_components.py

```diff
@@ -186,6 +186,50 @@ def _generate_output_file_name(port_name):
     return _outputs_dir + '/' + _sanitize_file_name(port_name) + '/' + _single_io_file_name
 
 
+# Placeholder to represent the output directory hosting all the generated URIs.
+# Its actual value will be specified during pipeline compilation.
+OUTPUT_DIR_PLACEHOLDER = '{{kfp.output_dir}}'
+# Format of the Argo parameter used to pass the producer's Pod ID to
+# the consumer.
+PRODUCER_POD_NAME_PARAMETER = '{}-producer-pod-id-'
+
+
+def _generate_output_uri(port_name: str) -> str:
+    """Generates a unique URI for an output.
+
+    Args:
+        port_name: The name of the output associated with this URI.
+
+    Returns:
+        The URI assigned to this output, which is unique within the pipeline.
+    """
+    return os.path.join(
+        OUTPUT_DIR_PLACEHOLDER,
+        '{{workflow.uid}}',
+        '{{pod.name}}',
+        port_name
+    )
+
+
+def _generate_input_uri(port_name: str) -> str:
+    """Generates the URI for an input.
+
+    Args:
+        port_name: The name of the input associated with this URI.
+
+    Returns:
+        The URI assigned to this input, will be consistent with the URI where
+        the actual content is written after compilation.
+    """
+    return os.path.join(
+        OUTPUT_DIR_PLACEHOLDER,
+        '{{workflow.uid}}',
```

Comment:
Are you sure we want to make the core of the SDK to Argo-specific?

Reply:
Good point. An easy improvement could be making this a plain placeholder and if compiled on Argo dsl.RUN_ID will be swapped in.